synapse/handlers/search.py

```diff
@@ -418,13 +427,10 @@ async def search(self, user, content, batch=None):
 
         state_results = {}
         if include_state:
-            rooms = {e.room_id for e in allowed_events}
-            for room_id in rooms:
+            for room_id in {e.room_id for e in allowed_events}:
                 state = await self.state_handler.get_current_state(room_id)
                 state_results[room_id] = list(state.values())
 
-            state_results.values()
```

Comment:
I don't think this was doing anything, but maybe there was *magic* it was doing?

Reply:
That looks spurious to me. `Dict.values` shouldn't have any side effect.